src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -789,12 +789,15 @@ bool QgsPostgresProvider::loadFields()
   // Clear cached information about enum values support
   mShared->clearSupportsEnumValuesCache();
 
+  QString sql;
+  QString attroidsFilter;
+
   if ( !mIsQuery )
   {
     QgsDebugMsgLevel( QStringLiteral( "Loading fields for table %1" ).arg( mTableName ), 2 );
 
     // Get the relation oid for use in later queries
-    QString sql = QStringLiteral( "SELECT regclass(%1)::oid" ).arg( quotedValue( mQuery ) );
+    sql = QStringLiteral( "SELECT regclass(%1)::oid" ).arg( quotedValue( mQuery ) );
```

Comment:
Unrelated: for performance reasons, could we do

```sql
QString tableoid = QStringLiteral( "regclass(%1)::oid" ).arg( quotedValue( mQuery ) );
```

and completely remove one query (roundtrip)?

Reply:
Ok, squeezed into one query (although initially I wanted to change as less as possible, to avoid any unnecessary backport risk ;) )